authorization/usecase/core.go

```diff
@@ -2,26 +2,47 @@ package usecase
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"log/slog"
 	"math/rand"
 	"regexp"
 	"sync"
 	"time"
 
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/authorization/repository/csrf"
 	"github.com/go-park-mail-ru/2023_2_Vkladyshi/authorization/repository/profile"
 	"github.com/go-park-mail-ru/2023_2_Vkladyshi/authorization/repository/session"
 	"github.com/go-park-mail-ru/2023_2_Vkladyshi/configs"
-	"github.com/go-park-mail-ru/2023_2_Vkladyshi/errors"
+	"github.com/go-park-mail-ru/2023_2_Vkladyshi/pkg/models"
 )
 
+type ICore interface {
+	CreateSession(ctx context.Context, login string) (string, session.Session, error)
+	KillSession(ctx context.Context, sid string) error
+	FindActiveSession(ctx context.Context, sid string) (bool, error)
+	CreateUserAccount(login string, password string, name string, birthDate string, email string) error
+	FindUserAccount(login string, password string) (*models.UserItem, bool, error)
+	FindUserByLogin(login string) (bool, error)
+	GetUserName(ctx context.Context, sid string) (string, error)
+	GetUserProfile(login string) (*models.UserItem, error)
+	EditProfile(prevLogin string, login string, password string, email string, birthDate string, photo string) error
+	CheckCsrfToken(ctx context.Context, token string) (bool, error)
+	CreateCsrfToken(ctx context.Context) (string, error)
+	CheckPassword(login string, password string) (bool, error)
+}
+
 type Core struct {
 	sessions   session.SessionRepo
 	mutex      sync.RWMutex
 	lg         *slog.Logger
 	users      profile.IUserRepo
+	csrfTokens csrf.CsrfRepo
 }
 
+var InvalideEmail = errors.New("invalide email")
+var LostConnection = errors.New("Redis connection lost")
```

Comment:
В гошке принято давать префик Err переменным-ошибкам, чтобы отличать их от остальных

```suggestion
var ErrInvalideEmail = errors.New("invalide email")
var ErrLostConnection = errors.New("Redis connection lost")
```